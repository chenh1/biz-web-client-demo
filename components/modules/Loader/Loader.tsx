import React from "react"
import { Box } from "../../containers/Box/Box";
import './loader.css'

interface LoaderProps {
  width?: number;
  height?: number;
}

export const Loader: React.FC<LoaderProps> = ({ width = 256, height = 256 }: LoaderProps) => {
  const baseUrl = process.env.STATIC_URL || 'https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com'
  return (
    <Box padding="0" className="slow-spin justify-center">
      <img src={`${baseUrl}/biz-web-client-demobloommono_ba249abb01.png`} width={width} height={height} alt="Logo icon"/>
    </Box>
  )
}