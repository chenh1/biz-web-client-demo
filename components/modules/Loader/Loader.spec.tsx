import { Loader } from "./Loader";
import { render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders Loader module', async () => {
  const { getByTestId, asFragment } = render(
    <Loader/>
  )

  expect(asFragment()).toMatchSnapshot();
});
