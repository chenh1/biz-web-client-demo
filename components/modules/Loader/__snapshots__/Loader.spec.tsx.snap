// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader module 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-4  p-2 slow-spin justify-center"
  >
    <img
      alt="Logo icon"
      height="256"
      src="https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com/biz-web-client-demobloommono_ba249abb01.png"
      width="256"
    />
  </div>
</DocumentFragment>
`;
